### PR TITLE
fix(ENG-1871): 404 path in the url should return 404 status

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
The links in the report are showing a 404 page, but the HTTP status code of this page is 200, for example:
- /._darcs
- /.bzr
- /.hg
- /BitKeeper

## Linear Ticket
https://linear.app/autokitteh/issue/ENG-1871/casa-dast-scan-1-hidden-files-found

## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [x] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/pages/external404.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+import { Error404 } from "@assets/image";
+
+export const External404 = () => {
+	return (
+		<div className="flex w-full flex-col items-center justify-center py-5">
+			<Error404 className="w-1/3" />
+
+			<div className="mt-16 font-fira-code text-lg text-black">Error: Page doesn&apos;t exist</div>
```
Comment (J1za) [inline src/components/pages/external404.tsx:10]: add to 18n

File: src/components/pages/external404.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+			<a className="mt-4 font-fira-code text-lg font-bold text-black" href="/">
+				Go to Homepage
```
Comment (J1za) [inline src/components/pages/external404.tsx:12]: add to 18n

File: src/components/templates/descopeMiddleware.tsx [hunk 3]
```diff
@@ -84,13 +96,24 @@ export const DescopeMiddleware = ({ children }: { children: ReactNode }) => {
...
 	}
 
+	if (location.pathname === "/404") {
+		return <External404 />;
+	}
+
+	const matches = matchRoutes(routes, location);
+
+	if (!matches) {
+		window.location.href = "/404";
```
Comment (J1za) [inline src/components/templates/descopeMiddleware.tsx:112]: We can use `useNavigate` ?

File: src/components/pages/external404.tsx (hunk 1 above)
Comment (RonenMars) [reply src/components/pages/external404.tsx:10]: added

File: src/components/pages/external404.tsx (hunk 2 above)
Comment (RonenMars) [reply src/components/pages/external404.tsx:12]: added

File: src/components/templates/descopeMiddleware.tsx (hunk 3 above)
Comment (RonenMars) [reply src/components/templates/descopeMiddleware.tsx:112]: Yes, changed

File: src/components/pages/external404.tsx (hunk 1 above)
Comment (RonenMars) [reply src/components/pages/external404.tsx]: done

**Pull request merged**